plugins/catalog/src/components/CatalogPage/CatalogPage.tsx

```diff
@@ -121,7 +125,16 @@ const CatalogPageContents = () => {
             id: 'owned',
             label: 'Owned',
             icon: SettingsIcon,
-            filterFn: entity => entity.spec?.owner === userId,
+            filterFn: entity => {
+              const ownerRelation = entity.relations?.find(
```

Comment:
Might be a candidate for a hook, since this is a quite common requirement.

Reply:
Yeah see below for the comment about `useUserRelations` as well.

BUT, I am leaning towards it being better if no code assumes the number of relations. For example, even if there's only one owner in the `spec`, it's better if the code never assumes that there'll always and forever only be one owner relation. So I think it would make sense if this didn't use `find`, but rather `filter`, and then write the loop necessary to match all of the owners with all of the memberships. Sounds like a double for loop is necessary.

Also, what if the owner relation was `user:default/freben`? The set of things to check against probably includes the user as well as their memberOf groups.

This could probably be wrapped in a helper function `userIsOwnerOf(user: Entity, target: Entity)`

Reply:
I assumed that the ownership was only valid for groups, but I should've read the docs more thoroughly:

> An ownership relation where the owner is usually an organizational entity (User or Group), and the other entity can be anything.

I'll update the PR!